### PR TITLE
Force remeasures on font timeout/load

### LINKED ISSUE #10195: amp-list appears not to load on Safari in some cases
From @nzeeshan

> I am using amp-mustache to render a list of links on my amp page. It works fine on Android devices, however on Safari iOS, it doesn't work. Here's a [screenshot](http://imgur.com/a/ClqJb) of what I get when accessing the page through Safari.

[AMP page URL](https://www.bnymellon.com/us/en/our-thinking/amp/swimming-against-the-tide.jsp)
[JSON URL](https://www.bnymellon.com/us/en/our-thinking/amp/frances.json)

I am using full paths, not even relative paths and not sure how to resolve this. What am I doing wrong and why does it only work on Android?

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #10195.

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -254,6 +258,22 @@ export class Resources {
       this.schedulePass();
       this.monitorInput_();
     });
+
+    // Safari 10 and under incorrectly estimates font spacing for `@font-face`
```
Comment (jridgewell) [inline src/service/resources-impl.js]: Can we guard this so only Safari does it?

Comment (dvoytenko) [inline src/service/resources-impl.js]: We could. Overall, this is not a horrible idea and expense is likely small - we only remeasure things here. Do you think it's worth to guard this by Safari only? If so, I'll add.

Comment (jridgewell) [inline src/service/resources-impl.js]: Nah, this if fine as is. Why 3100 ms?

Comment (dvoytenko) [inline src/service/resources-impl.js]: "Documented" value is 3000ms. In all my tests I got 3050ms. So, I added a bit on top to 3100ms. Kind of really "rule of thumb" thing, but the idea is to get a "fair effort" to remeasure.

Comment (jridgewell) [inline src/service/resources-impl.js]: 👍

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -254,6 +258,22 @@ export class Resources {
...
+    // fonts. This leads to wild measurement errors. The best course of action
+    // is to remeasure everything on window.onload or font timeout (3s),
+    // whichever is earlier. This has to be done on the global window because
+    // this is where the fonts are always added. Unfortunately,
+    // `document.fonts.ready` cannot be used here due to
+    // https://bugs.webkit.org/show_bug.cgi?id=174030.
+    // See https://bugs.webkit.org/show_bug.cgi?id=174031 for more details.
+    Promise.race([
+      loadPromise(this.win),
+      timerFor(this.win).promise(3100),
```
Comment (cramforce) [inline src/service/resources-impl.js]: It would be saver to start waiting at doc-ready. This might be before fonts matched.

**Pull request merged**